src/components/CellsContainer/helpers.js

```diff
@@ -0,0 +1,92 @@
+import { create } from 'd3';
+
+export function getCellSize(totalCellsCount) {
+  switch (true) {
+    case totalCellsCount < 100:
+      return 38;
+
+    case totalCellsCount < 500:
+      return 20;
+
+    case totalCellsCount < 1000:
+      return 18;
+
+    case totalCellsCount < 2000:
+      return 16;
+
+    default:
+      return 11;
+  }
+}
+
+export function getCellProps(index, itemsCount) {
+  const [lived, total] = itemsCount;
+  const color = index < lived ? 'IndianRed' : 'SteelBlue';
+
+  return {
+    color,
+    tooltipText: 'Number: ' + ++index
+  };
+}
+
+export function setAttributes(el, attrs) {
+  Object.keys(attrs).forEach((attr) => {
+    el.attr(attr, attrs[attr]);
+  });
+}
+
+export const createSvg = (width, height) =>
+  create('svg').attr('width', width).attr('height', height);
+
+export const getId = (name, index) => `${name}-${index}`;
+
+export function createRectsData({
+  containerWidth,
+  count,
+  size,
+  gap,
+  getAdditionalRectProps,
+  color = 'SteelBlue',
+  hoverColor = 'IndianRed'
+}) {
+  const emptyData = [...Array(count)];
+  if (!containerWidth) return emptyData;
+
+  const rectSize = size + gap * 2;
+  const colsAmount = Math.floor(containerWidth / rectSize);
+  const rowsAmount = Math.ceil(count / colsAmount);
+  const xOffset = (containerWidth - rectSize * colsAmount) / 2 + gap;
+
+  return emptyData.map((_, i) => {
+    const additionalProps =
+      typeof getAdditionalRectProps === 'function'
```

Comment:
why it should be a function?
you can pass just object and if you need a function to create such object, would be better to create it outside the component

Reply:
Because we need to create as many objects as our count value.
Do you mean moving map function outside?